feat: dir_display_limit

kawre opened the pull request:
New feature to limit the number of directories to display.
Especially useful when working with something like Spring Boot, where just the directory name takes more than 50% of my screen real estate.

utilyre commented on lua/barbecue/ui/components.lua:
```diff
@@ -47,11 +47,16 @@ function M.dirname(bufnr)
 
   local dirs = vim.split(dirname, PATH_SEPARATOR, { trimempty = true })
-  for _, dir in ipairs(dirs) do
+  local dir_display_limit = math.floor(config.user.dir_display_limit)
```
Set `@type` to `integer` in the `template.lua` file and remove this call.

kawre commented:
Changed the type to integer and added ellipsis symbol to indicate that the directories have been split. Also unsure how to handle the protocol prefix. Do you have any ideas?

utilyre commented:
Honestly, it's been a while since I used this plugin myself. So just be sure that nothing breaks.

Cases that I recommend testing:

1. Relative file paths (especially when there is a `./some.txt` or `../some.txt`)
2. Absolute file paths (they should have a `/` entry at the beginning)
3. URLs should have a `http://` entry at the beginning (http is just an example it should handle any given schema like `tcp://`, `https://`, or `mycustomschema://`)
